### DISCUSSION TITLE
Using gqlgen for generating Federation Gateway

### DISCUSSION THREAD

**Discussion by patilakshay227** [99designs/gqlgen#1981] [General]
With the Existing gqlgen federation, we need to use the apollo server as a gateway layer which is implemented in NodeJS. 
We have observed the Go service providing better performance as compared to equivalent NodeJS service. 

**Can we generate the gateway itself in Go using gqlgen?** 

Basically no dependency on Node (Below line shouldn't be needed)
```
npm install --save @apollo/gateway apollo-server graphql
```

**Comment by fasibio** [comment]
Do you know this ? 
https://github.com/jensneuse/graphql-go-tools#apollo-federation-gateway-replacement

**Reply by patilakshay227** [reply]
Thanks @fasibio

**Comment by giautm** [comment]
Or you can use [`router`](https://github.com/apollographql/router). It was written in Rust.
